lib/classes/config-schema-handler/resolve-ajv-validate.js

```diff
@@ -67,6 +68,10 @@ const getValidate = async (schema) => {
     loadedModuleCode,
     path.resolve(__dirname, `[generated-ajv-validate]${filename}`)
   );
+  if (typeof validator !== 'function') {
+    log.debug(`Invalid validator: ${validator}`);
+    log.debug(`Code loaded from cache path: ${loadedModuleCode}`);
```

Comment:
Debug logs are good in _success_ paths, as they're hidden by default, and we can turn on chosen when needed.

Here we're in context of WTF moment, when we know things are wrong and we need to throw programmer error.

In such case I would just write with error level (it's important that if user reports that on forum, this logs are shown, with `log.debug` they won't)

I would also not coerce `validator` to string (it can turn to `[object Object]`), it's way better to log it directly as object (internally sophisticated inspection is used, that may expose important information we won't see if it's stringified)

```javascript
log.error("Unexpected validator %o, resolved from source %s", validator, loadedModuleCode);

throw new Error("Unexpected non-function AJV validator type. Please report at https://github.com/serverless/serverless including all the logs output");
```

Reply:
Great suggestion - thanks

Reply:
You can put all of that into the `Error` message directly as well, that will avoid having 2 different error messages in the CLI output.

Reply:
I think the error might coerce the object and produce non-full output - I'll leave it this way as it's only intended to be temporary (hopefully) after we get to the root of this issue as it happens randomly according to report on forum

Reply:
👍